### PR TITLE
Eaton 5P / 5PX UPS Issues #3364 #3508

### PR DESCRIPTION
**Pull request by oam7575**
Updates related to the detection of lower end Eaton UPS devices.

Confusion is caused by the Eaton Network Management Cards supporting **both** EATON and old ex-MGE MIB oids.

detection for both Eaton and probably some MGE UPS's should be improved.

I will attempt to add more sensors in further updates.

Note: first attempt at this - be gentle - let me know if i need to fix formatting - tidy up etc.

### CODE REVIEW

File: includes/discovery/sensors/voltages/eatonups.inc.php [hunk 1]
```diff
@@ -27,14 +27,14 @@
     }
 
     // XUPS-MIB::xupsInputNumPhases.0 = INTEGER: 1
-    $oids = trim(snmp_walk($device, 'xupsInputNumPhases', '-OsqnU', 'XUPS-MIB'));
+    $oids = trim(snmp_walk($device, '.1.3.6.1.4.1.534.1.3.4.1.1', '-OsqnU', 'XUPS-MIB'));
```
Comment (laf) [inline includes/discovery/sensors/voltages/eatonups.inc.php]: How come you've changed this to the numeric OID?

File: includes/discovery/sensors/voltages/eatonups.inc.php [hunk 2]
```diff
@@ -27,14 +27,14 @@
...
     d_echo($oids."\n");
 
     list($unused,$numPhase) = explode(' ', $oids);
     for ($i = 1; $i <= $numPhase; $i++) {
         // XUPS-MIB::xupsInputVoltage.1 = INTEGER: 228
         $volt_oid = ".1.3.6.1.4.1.534.1.3.4.1.2.$i";
-        $descr    = 'Output';
+        $descr    = 'Input Voltage';
```
Comment (laf) [inline includes/discovery/sensors/voltages/eatonups.inc.php]: Is the label Output incorrect?

File: includes/discovery/sensors/voltages/eatonups.inc.php [hunk 3]
```diff
@@ -48,14 +48,14 @@
     }
 
     // XUPS-MIB::xupsOutputNumPhases.0 = INTEGER: 1
-    $oids = trim(snmp_walk($device, 'xupsOutputNumPhases', '-OsqnU'));
+    $oids = trim(snmp_walk($device, '.1.3.6.1.4.1.534.1.4.4.1.1', '-OsqnU'));
```
Comment (laf) [inline includes/discovery/sensors/voltages/eatonups.inc.php]: How come you've changed this to the numeric OID?

File: includes/discovery/sensors/voltages/eatonups.inc.php [hunk 4]
```diff
@@ -68,6 +68,28 @@
...
+        $watts_oid = ".1.3.6.1.4.1.534.1.4.4.1.4.$i";
+        $descr    = 'Output Watts';
+        if ($numPhase > 1) {
+            $descr .= " Phase $i";
+        }
+
+        $type    = 'xups';
+        $divisor = 1;
+        $power = (snmp_get($device, $watts_oid, '-Oqv') / $divisor);
+        $index   = '4.4.1.4.'.$i;
```
Comment (laf) [inline includes/discovery/sensors/voltages/eatonups.inc.php]: I'd suggest sticking with the full oid rather than partial.

**Comment by laf** [comment]
Mostly looks good, few comments inline.

Can you also take a look at your indentation, it's a bit all over in some places. We use four spaces for a tab.

You also have a lot of commented out code, please just remove this.

**Comment by oam7575** [comment]
@laf 

Thanks for checking and the feed back.
Notes:
1 - I am very new to all of this
2 - Based on further reading of LibreNMS docs I have been attempting to split some of the readings into seperate config files
3 - I have been hacking round a little more trying to tidy stuff up.
4 - Feel free to reject this if need be - either way I will be attempting to submit a cleaner / better patch shortly.

Cheers.

File: includes/discovery/sensors/voltages/mgeups.inc.php [hunk 5]
```diff
@@ -52,4 +52,65 @@
...
+   }
+
+    $ups_battery_level_oid = '.1.3.6.1.4.1.705.1.5.2.0';
+    $ups_battery_level     = snmp_get($device, $ups_battery_level_oid, '-Oqv');
+
+    if (is_numeric($ups_battery_level)) {
+        discover_sensor($valid['sensor'], 'charge', $device, $ups_battery_level_oid, 'UPS Battery Level', $ups_device_manufacturer.' '.$ups_device_model, 'UPS Battery Level', '1', '1', null, null, null, null, $ups_battery_level);
+    }
+
+//    $ups_battery_remaining_time_oid = '.1.3.6.1.4.1.705.1.5.1.0';
```
Comment (murrant) [inline includes/discovery/sensors/voltages/mgeups.inc.php]: Still some commented code left to remove.  Thanks.

**Comment by murrant** [comment]
Looks pretty close to being commit-able, just some commented code to remove.  Also, we can squash on merge if we there there are too many commits, so no worries there.

**Comment by oam7575** [comment]
@murrant 
Thanks for the feed back.

I am still very confused by how git works so that is making things a whole lot messier than should be.

At this stage I have attempted to
* up the formatting and comments 
* push to my dev branch
* merge my dev branch into my master 

Let me know what you think.

Cheers.

**Comment by murrant** [comment]
FYI, merging your dev branch into your master may cause you issues later.  I try to leave my master the same as upstream.

**Comment by laf** [comment]
Yeah this needs tidying up a little.

You'll probably be better to create a branch from your current master which will contain all of these changes (git checkout -b issue-3364). Then squash all of the commits into one with something like git rebase --interactive HEAD~35. In that file, change all but one pick word to squash. Save the file and you should now have a 1 commit branch.

Push that, create a new PR from it and close this down.

Go back to your master (git checkout master) and reset it back to the origin (assuming our repo is called origin for you do git reset --hard origin/master

**Comment by oam7575** [comment]
@laf 

Thanks for the suggestion.
As I am sure you have noticed I am a little confused by how git works.

I'll give it a shot and see were we land.

Cheers.

**Comment by oam7575** [comment]
Trying to work through the mess I have created.
I will close this now and try to start from scratch.

Sorry for the mess - but thanks for the tips and your patience with me.

Cheers.

**Pull request closed without merging**